ui/src/components/imquery/EditMatch.vue

```diff
@@ -0,0 +1,348 @@
+<template>
+  <div class="edit-match-wrapper">
+    <Button
+      :severity="editMatch.exclude ? 'danger' : 'secondary'"
+      :outlined="!editMatch.exclude"
+      label="NOT"
+      @click="
```

Comment:
use @click.stop notation to stop propagation. https://vuejs.org/guide/essentials/event-handling